tests/atlocal.in

```diff
@@ -110,37 +110,23 @@ setup_env()
     fi
 }
 
-run()
-{
-    setup_env
-    "$@" --define "_buildhost testhost" \
-         --define "_tmppath ${RPMTEST}/tmp" \
-         --define "_topdir ${RPMTEST}/build" \
-         --define "_db_backend ${DBFORMAT}" \
-         --dbpath="${RPMTEST}/var/lib/rpm/"
-}
-
 rundebug()
 {
-    setup_env
     export RPM_CONFIGDIR=${RPMTEST}/@RPM_CONFIGDIR@
     cp ${RPMDATA}/macros.debug ${RPM_CONFIGDIR}/macros.d/
-    run "$@"
+    "$@"
     rm -f ${RPM_CONFIGDIR}/macros.d/macros.debug
     unset RPM_CONFIGDIR
 }
 
 runroot()
 {
-    setup_env
     snapshot exec "$@" \
-             --define "_buildhost testhost" \
-             --define "_topdir /build"
+             --define "_tmppath /var/tmp"
```

Comment:
This is indeed strange that `runroot` doesn't work with `/tmp` as `_tmppath`, I've looked at it closer but couldn't find the culprit, it's probably related to some `tmpfs` foo, as you've noted in the commit message.

Nevertheless, we might as well just set `_tmppath` to `/var/tmp` in `macros.testenv` too which would eliminate the need for this override here.

Reply:
The problem is that /var is read-only by default, and rpm needs temporary files for all sorts of things that don't otherwise require writing to the root image.

Reply:
Eh, or that's what I thought, but apparently not.

Reply:
Yep, I realized that `/var/tmp` shouldn't be writable after writing the above comment, too :sweat_smile: And indeed, it *is* writable for some reason, lol... Mystery deepens...

Reply:
Oh, right. So `/var/tmp` is r/w even in `podman --read-only` mode, see the man page for `--read-only` :smile:

Reply:
That's handy, if it actually works for us. It seems to, locally, whereas /tmp didn't. I'll push that change to the end to see if it also works on Ubuntu, but I wouldn't make this a blocker here - there's always the next rainy day to clean up further.

Reply:
So using `/var/tmp` would work for the "native" (originally `run`) case, too, however it would violate the rule that no test can influence other tests (since now they would all share a common `/var/tmp`). I guess let's leave it as is for now, then. It could be managed somehow, pretty sure, just not something to block this PR on.

Reply:
And, it may be writable on podman but apparently not docker. I'll just unpush the last commit, lets forget that part for now.

Reply:
Yep, it's tmpfs, but there's only one instance of a podman container for the whole test-suite. `make atshell` is equivalent to running `make check` with only one test-case :smile:

Reply:
... but since Ubuntu was now failing, for a change... let's leave it for later, indeed :smile:

Reply:
Right. Anyway, lets leave that to a future exercise :laughing: